src/main/java/com/google/sps/servlets/MealServlet.java

```diff
@@ -1,47 +1,117 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
-
 package com.google.sps.servlets;
 
 import java.io.IOException;
-import javax.servlet.annotation.WebServlet;
-import java.util.Arrays;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.stream.Collectors;
+import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import com.google.gson.Gson;
 import com.google.sps.data.Meal;
+import java.util.HashMap;
 
-@WebServlet("/meal")
+@WebServlet("/meal/*")
 public class MealServlet extends HttpServlet {
 
-  Meal meal;
+    private final HashMap < Long, Meal > dishes = new HashMap < Long, Meal > ();
+
+    @Override
+    public void init() {
+        // TODO(sandatsian): implement uploading of dataset from website here
+        dishes.put(0L, new Meal(0L, "fried potato", "fried potato with mushrooms and onion",
+            new ArrayList < > (Arrays.asList("potato", "onion", "mushrooms", "oil")), "Main"));
+        dishes.put(1L, new Meal(1L, "Italian pizza", "Pizza with pineaple, sausage and tomato",
+            new ArrayList < > (Arrays.asList("flour", "water", "sausage", "tomato", "pineaple", "cheese")), "pizza"));
+        dishes.put(2L, new Meal(2L, "vegetable soup", "vegetable soup with onion",
+            new ArrayList < > (Arrays.asList("potato", "onion", "cabbage", "mushrooms", "water", "carrot", "pumpkin")), "soup"));
+        dishes.put(3L, new Meal(3L, "chocolate cake", "chocolate cake with butter cream and strawberry",
+            new ArrayList < > (Arrays.asList("flour", "water", "butter", "cacao", "chocolate", "sugar", "strawberry", "eggs")), "dessert"));
+        dishes.put(4L, new Meal(4L, "hot chocolate", "hot chocolate with sugar and caramel",
+            new ArrayList < > (Arrays.asList("cacao", "sugar", "milk", "caramel", "vanil")), "drinks"));
+    }
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String pathInfo = request.getPathInfo();
+
+        // Get meal
+        // GET/meal/
+        if (pathInfo == null || pathInfo.equals("/")) {
+            searchMeal(request, response);
+            return;
+        }
+
+        // GET/meal/*/*
+        if (pathInfo.split("/").length != 2) {
+            response.sendError(HttpServletResponse.SC_BAD_REQUEST);
+            return;
+        }
+
+        // GET/meal/meal_id
+        if (pathInfo.split("/").length == 2) {
+            getMealById(request, response);
+            return;
+        }
+
+        response.sendError(HttpServletResponse.SC_NOT_FOUND);
+        return;
+    }
+
+    private void searchMeal(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String gson = new Gson().toJson(dishes);
```

Comment:
Add a `TODO(polinasand)` comment that says this is just a hardcoded result, and that you'll replace it with actual search in the future. Also specify whether you'll add search for prototype or only later, for MVP.